nameable.rb

```diff
@@ -0,0 +1,39 @@
+require_relative 'person'
+
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'not implemented'
+  end
+end
+
+class BaseDecorator < Nameable
+  atrr_accessor :nameable
+
+  def initilize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitilizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitilize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name[0, 10]
```

Comment:
- [ ] I think this method should have a condition since in the requirments that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word. so you should add an if else condition so check the lenght first if it is more than 10 char so trim if not just return the name ✨💯🚀

- Hint: when you want to trim I think there is a best way to do it [0..9] try this 🚀💯✨

![image](https://user-images.githubusercontent.com/58553711/210457316-26776441-9696-40e2-aee8-dd55d4b5edb8.png)

Reply:
@Reem-lab Thank you for Hint